### PR TITLE
Created Related Items section

### PR DESCRIPTION
**Pull request by andreaWaldren**
**JIRA Ticket**: (link) (:star:)
https://webapps.es.vt.edu/jira/browse/LIBTD-2152

# What does this Pull Request do? (:star:)
Creates a section for Related items which will show at the bottom of the archive page. This section first searches for other items in the same subcollection. If there are less than ten, then it will search for items in the direct parent collection of this subcollection, and the items from the other subcollections of the parent collection. Once it finds 10 items total it will stop building a list and display the results. If it doesn't find 10 items, then it will return whatever number of items were found. If no other items are found then then section won't show.

# What's the changes? (:star:)
-related_items_spec.js - cypress test for this component
-package.json - the carousel relies on two npm packages, react-slick and slick-carousel (https://react-slick.neostack.com/)
-RelatedItems.js and RelatedItems.css - all the code for the component
-ArchivePage.js  - adds the component to the archive page.

# How should this be tested?
-Use related_items_spec.js to run the cypress test.

# Interested parties
@yinlinchen 

(:star:) Required fields

### CODE REVIEW

File: src/components/RelatedItems.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    );
+    return uniqueItems;
+  }
+
+  removeItem(item_list) {
+    let arr = item_list.filter(
+      value => value.identifier !== this.props.collection.identifier
+    );
+    return arr;
+  }
```
Comment (yinlinchen) [inline src/components/RelatedItems.js:48]: Can you merge these two functions into one?

File: src/components/RelatedItems.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    let newArray = [];
+    if (item_list.length > 10) {
+      newArray = item_list.slice(0, 10);
+    } else {
+      newArray = item_list;
+    }
+
+    if (newArray.length > 0) {
```
Comment (yinlinchen) [inline src/components/RelatedItems.js:71]: Can it be `if (item_list.slice(0, 10).length > 0)` and remove line 64-68 ?

File: src/components/RelatedItems.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      i--;
+      limit = limit + 10;
+    }
+
+    let newArray = [];
+    if (item_list.length > 10) {
+      newArray = item_list.slice(0, 10);
+    } else {
+      newArray = item_list;
+    }
```
Comment (yinlinchen) [inline src/components/RelatedItems.js:69]: let newArray = item_list.slice(0, 10)

File: src/components/RelatedItems.js (hunk 2 above)
Comment (yinlinchen) [inline src/components/RelatedItems.js:71]: relatedItemsArray.length

File: src/components/RelatedItems.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+    if (newArray.length > 0) {
+      this.setState({
+        items: newArray
```
Comment (yinlinchen) [inline src/components/RelatedItems.js:73]: relatedItemsArray

File: src/components/RelatedItems.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      });
+    }
+  }
```
Comment (yinlinchen) [inline src/components/RelatedItems.js:76]:
```
     this.setState({
        items: item_list.slice(0, 10)
      });
```

**Review by yinlinchen — CHANGES_REQUESTED** [review]
Please see comments

File: src/components/RelatedItems.js (hunk 1 above)
Comment (andreaWaldren) [reply src/components/RelatedItems.js:48]: Yes!

File: src/components/RelatedItems.js (hunk 5 above)
Comment (andreaWaldren) [reply src/components/RelatedItems.js]: I fixed this too!

**Review by yinlinchen — APPROVED** [review]
Nicely done! Thanks for the hard working!

**Pull request merged**